src/Test/WebTestCase.php

```diff
@@ -224,22 +226,60 @@ protected function getDependencyInjectionContainer(): ContainerInterface
         return $this->containers[$cacheKey];
     }
 
+    protected static function createKernel(array $options = []): KernelInterface
+    {
+        if (!isset($options['environment'])) {
+            $options['environment'] = self::$env;
+        }
+
+        return parent::createKernel($options);
+    }
+
     /**
      * Keep support of Symfony < 5.3.
      */
     public function __call(string $name, $arguments)
     {
         if ('getContainer' === $name) {
-            if (method_exists($this, $name)) {
-                return self::getContainer();
-            }
-
             return $this->getDependencyInjectionContainer();
         }
 
         throw new \Exception("Method {$name} is not supported.");
     }
 
+    public function __set($name, $value)
+    {
+        if ($name !== 'environment') {
+            throw new \Exception(sprintf('There is no property with name "%s"', $name));
```

Comment:
What if I extend WebTestCase and I use `$this->client` (or whatever new variable), will it throw an exception?

Reply:
Actually some tests already declare new variables (e.g. [`$this->client =`](https://github.com/liip/LiipFunctionalTestBundle/pull/593/files#diff-2a2f8a611efd3fe4ee275790ac720835aab87fdcef0ba731d3754e40a3b92f01)) but I don't understand why it doesn't trigger this exception.

Reply:
This code is executed only when the variable does not exist. See https://www.php.net/manual/en/language.oop5.overloading.php#object.set